Add interface option for disabling city view highlighting of plots buffed by hovered building

artemigkh opened the pull request:
Adds an option for disabling highlighting of plots that would be buffed by a building when hovered over in the construction menu
![image](https://github.com/LoneGazebo/Community-Patch-DLL/assets/8049171/eedf3d23-d799-47d2-b087-da9521efe4c8)

This change should provide a workaround for people experiencing some city view CTDs as it appears that `SerialEventHexHighlight` may cause memory problems

* https://github.com/LoneGazebo/Community-Patch-DLL/issues/10534
* https://github.com/LoneGazebo/Community-Patch-DLL/issues/10574
* https://github.com/LoneGazebo/Community-Patch-DLL/issues/10848
* https://forums.civfanatics.com/threads/new-version-4-7-3-april-28-2024.689478/page-2#post-16598238

azum4roll commented:
It's unlikely to be SerialEventHexHighlight since it's used by worked plots as well and has never caused issues.

artemigkh commented:
I think it's because of the way it interacts with the existing city hex highlights, as Squads also uses highlights extensively and I've never heard of any problems there

azum4roll commented:
So technically on EUI if you move your mouse on and off cities quickly it'll also crash.

axatin commented:
> If it's causing memory crashes, shouldn't we just remove it entirely?

or at least make it an option that’s turned off by default

artemigkh commented:
I tested it and `SerialEventHexHighlight` also reliably causes crashes outside of city view. Turning off `SerialEventHexHighlight` by default, even though helps for some people (I will note even before the addition of building bonus highlights, people had memory CTDs with EUI that were resolved by removing it, probably in part from the very same plot highlighting functionality) is a solution that is plainly throwing away the baby with the bathwater. Plot highlights provide invaluable information in many facets of the game, and turning them off because a small minority of people have problems is a solution I'm not a fan of.

I have no plans for further changes to this MR; as it stands it still seems like a useful UI option for those that want more control over their display. If it is determined to be too much of a workaround that has no place in the community patch by its maintainers I totally understand, feel free to close it.

artemigkh commented:
After seeing several more reports of CTDs on building hover, I've changed my mind. Clearly a substantial number of people are consistently crashing on building selection list hover after the functionality I added, I'm closing this ticket and will shortly put up a new MR with it disabled by default and making it opt in instead.
